src/cast_core/include/cast_core/actors/SamplingLoader.hpp

```diff
@@ -36,21 +81,27 @@ namespace genny::actor {
 class SamplingLoader : public Actor {
 
 public:
-    explicit SamplingLoader(ActorContext& context);
+    SamplingLoader(ActorContext& context,
+                   std::string dbName,
+                   std::string collectionName,
+                   std::shared_ptr<DeferredSample> deferredSamplingLoader);
```

Comment:
Is it "better" to use `ShareableState`?

https://github.com/mongodb/genny/blob/bfaff11cabfa9e9f362458f597943259e754f1f1/src/cast_core/include/cast_core/actors/MonotonicSingleLoader.hpp#L81-L82

It's slightly before my morning coffee. And what you have is fine. But either way, consider a comment here about why this is done versus `ShareableState` so future copy/pastas know the right tradeoffs to consider.

Reply:
Uhh I am not understanding the point of SharableState... Maybe you can help? Is this to make it easier so that I don't have to add the custom actor construction code?

Reply:
Ok after starting to try it I determined it would not actually work, since we want to be able to use one sample per phase, but SharableState only supports one sample per workload. Added a comment.